karate-apache/src/main/java/com/intuit/karate/http/apache/ApacheHttpClient.java

```diff
@@ -327,8 +333,19 @@ protected HttpResponse makeHttpRequest(HttpEntity entity, ScenarioContext contex
             response.addCookie(cookie);
         }
         cookieStore.clear(); // we rely on the StepDefs for cookie 'persistence'
-        for (Header header : httpResponse.getAllHeaders()) {
-            response.addHeader(header.getName(), header.getValue());
+        for (Header header : httpResponse.getAllHeaders()) { // rely on setting cookies from set-cookie header, else these will be skipped.
+            if( header.getName().contains("Set-Cookie"))
```

Comment:
cookies are case insensitive

Reply:
got it, Thank you for the feedback - applied the review comment.